prac_09/files_to_sort_v2.py

```diff
@@ -0,0 +1,32 @@
+import os
+import shutil
+
+
+def main():
+    # Move to desired directory
+    os.chdir('FilesToSort')
+    print("Current directory is", os.getcwd())
+
+    for dir_name, subdir_list, file_list in os.walk('.'):
+        extensions_to_folders_dict = {}
+        for f in file_list:
+            extension = os.path.splitext(f)[-1]
+            key_exists = False
+            for key in extensions_to_folders_dict:
+                # If file extension already exists, skip creating overlapping category
+                if key == extension:
+                    key_exists = True
```

Comment:
What's going on here? You loop through each extension, but this will only remember/set `key_exists` for the final value in the loop.